code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm

```diff
@@ -0,0 +1,248 @@
+  */
+/datum/eldritch_knowledge/proc/recipe_snowflake_check(list/atoms,loc)
+	return TRUE
+
+/**
+  * What happens once the recipe is succesfully finished
+  *
+  * By default this proc creates atoms from result_atoms list. Override this is you want something else to happen.
+  */
+/datum/eldritch_knowledge/proc/on_finished_recipe(mob/living/user,list/atoms,loc)
+	if(result_atoms.len == 0)
+		return FALSE
+
+	for(var/X in result_atoms)
+		var/atom/A = X
+		new A(loc)
+
+	return TRUE
+
+/**
+  * Used atom cleanup
+  *
+  * Overide this proc if you dont want ALL ATOMS to be destroyed. useful in many situations.
+  */
+/datum/eldritch_knowledge/proc/cleanup_atoms(list/atoms)
+	for(var/X in atoms)
+		var/atom/A = X
+		qdel(A)
+	return
+
+/**
+  * Mansus grasp act
+  *
+  * Gives addtional effects to mansus grasp spell
+  */
+/datum/eldritch_knowledge/proc/mansus_grasp_act(atom/target, mob/user, proximity_flag, click_parameters)
+	return
+
+
+/**
+  * Sickly blade act
+  *
+  * Gives addtional effects to sickly blade weapon
+  */
+/datum/eldritch_knowledge/proc/eldritch_blade_act(atom/target, mob/user, proximity_flag, click_parameters)
+	return
+
+//////////////
+///Subtypes///
+//////////////
+
+/datum/eldritch_knowledge/spell
+	var/obj/effect/proc_holder/spell/spell_to_add
+
+/datum/eldritch_knowledge/spell/on_gain(mob/user)
+	var/obj/effect/proc_holder/S = new spell_to_add
+	user.mind.AddSpell(S)
+	. = ..()
+
+/datum/eldritch_knowledge/spell/on_lose(mob/user)
+	user.mind.RemoveSpell(spell_to_add)
+	. = ..()
+
+/datum/eldritch_knowledge/curse
+	var/timer = 5 MINUTES
+	var/list/fingerprints = list()
+
+/datum/eldritch_knowledge/curse/recipe_snowflake_check(list/atoms, loc)
+	fingerprints = list()
+	for(var/X in atoms)
+		var/atom/A = X
+		fingerprints |= A.return_fingerprints()
+	listclearnulls(fingerprints)
+	if(LAZYLEN(fingerprints) == 0)
+		return FALSE
+	return TRUE
+
+/datum/eldritch_knowledge/curse/on_finished_recipe(mob/living/user,list/atoms,loc)
+
+	var/list/compiled_list = list()
+
+	for(var/mob/living/carbon/human/H in GLOB.alive_mob_list)
+		if(fingerprints[md5(H.dna.uni_identity)])
+			compiled_list |= H.real_name
+			compiled_list[H.real_name] = H
+
+	if(LAZYLEN(compiled_list) == 0)
+		to_chat(user, "<span class='warning'>The items don't posses required fingerprints.</span>")
+		return
+
+	var/chosen_mob = input("Select the person you wish to curse","Your target") as null|anything in sortList(compiled_list, /proc/cmp_typepaths_asc)
+	if(!chosen_mob)
+		return ..()
+	curse(compiled_list[chosen_mob])
+	addtimer(CALLBACK(src, .proc/uncurse, compiled_list[chosen_mob]),timer)
+	. = ..()
+
+/datum/eldritch_knowledge/curse/proc/curse(mob/living/chosen_mob)
+	return
+
+/datum/eldritch_knowledge/curse/proc/uncurse(mob/living/chosen_mob)
+	if(!chosen_mob)
+		return
+
+/datum/eldritch_knowledge/summon
+	var/mob/living/mob_to_summon
```

Comment:
This is usually a path and that should be clear from the variable declaration unless you are doing some weird stuff with initial()

Reply:
Meaning to clear in `Destroy()`, @nemvar ?

Reply:
I just meant the declaration is weird.
This should just be ``var/mob_to_summon``, preferably with a comment above.

Reply:
i need to get real_name and name vars from the mob im summoning.